src/icmp/pinger.cc

```diff
@@ -233,14 +231,12 @@ main(int, char **)
             icmp4.Recv();
         }
 
-        if (PINGER_TIMEOUT + last_check_time < squid_curtime) {
+        if (PINGER_TIMEOUT <= timer.total().count()) {
             if (send(LINK_TO_SQUID, &tv, 0, 0) < 0) {
-                debugs(42, DBG_CRITICAL, "Closing. No requests in last " << PINGER_TIMEOUT << " seconds.");
+                debugs(42, DBG_CRITICAL, "Closing. No requests in last " << timer.total().count() << " seconds.");
```

Comment:
As detailed in another change request, count() is usually not "seconds". For one possible solution, see how `LFT_BUSY_TIME` handles a similar conversion to a known time unit. I have not tested this, but you may end up with something similar to:

```suggestion
                using seconds = std::chrono::duration<int, std::seconds>;
                debugs(42, DBG_CRITICAL, "Closing. No requests in last " << seconds(timer.total()).count() << " seconds.");
```

We can also add a ToHumanDuration() or a similar stream manipulator for printing std::chrono::duration interval duration values in informal messages:

```suggestion
                debugs(42, DBG_CRITICAL, "Closing. No requests during the last " << ToHumanDuration(timer.total()));
```

Reply:
Your suggestion results in type conversion operator errors with GCC-12.
I have gone with `duration_cast` in 4a88d2cdf13168e9e923db2c7564fec1ff9fd178

Reply:
> Your suggestion results in type conversion operator errors with GCC-12.

Yes, even if one fixes my (secondary) problems in the first sketch, the compiler _correctly_ prohibits the constructor-based conversion to seconds because that conversion is lossy. We _do_ want a lossy conversion here, but the compiler does not know that and stops us... Using lossy `duration_cast` is the right solution here AFAICT.